hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/recon/TestRecon.java

```diff
@@ -0,0 +1,395 @@
+    Thread.sleep(pauseInterval);
+
+    // HTTP call to /api/containers
+    containerResponse = makeHttpCall(containerKeyServiceURL);
+    map = new Gson().fromJson(containerResponse, HashMap.class);
+    linkedTreeMap = (LinkedTreeMap) map.get("data");
+    reconMetadataKeyCount = (long)(double)linkedTreeMap.get("totalCount");
+
+    //verify count of keys
+    Assert.assertEquals(omMetadataKeyCount, reconMetadataKeyCount);
+
+    //verify if Recon Metadata captures vol3/bucket3/key3 info in container3
+    containers = (ArrayList) linkedTreeMap.get("containers");
+    nestedContainerMap = (LinkedTreeMap)containers.get(3);
+    Assert.assertEquals(1.0, nestedContainerMap.get("NumberOfKeys"));
+    Assert.assertEquals(3, (long)(double)
+        nestedContainerMap.get("ContainerID"));
+
+    // HTTP call to /api/task/status
+    omLatestSeqNumber = ((RDBStore) metadataManager.getStore())
+        .getDb().getLatestSequenceNumber();
+
+    taskStatusResponse = makeHttpCall(taskStatusURL);
+    reconLatestSeqNumber = getReconMetadataKeyCount(taskStatusResponse, 2);
+
+    //verify sequence number after Delta Updates
+    Assert.assertEquals(omLatestSeqNumber, reconLatestSeqNumber);
+
+    File reconDbDir =
+        reconUtils.getReconDbDir(conf, OZONE_RECON_OM_SNAPSHOT_DB_DIR);
+    File lastKnownOMSnapshotBeforeRestart =
+        reconUtils.getLastKnownDB(reconDbDir, RECON_OM_SNAPSHOT_DB);
+
+
+    //restart Recon
+    cluster.restartReconServer();
+
+    //add 5 more keys to OM
+    addKeys(5, 10);
+    omKeyValueTableIterator = metadataManager.getKeyTable().iterator();
+    omMetadataKeyCount = getTableKeyCount(omKeyValueTableIterator);
+
+    //verify 10 keys in OM
+    Assert.assertEquals(10, omMetadataKeyCount);
+
+    //pause to get the next snapshot from om
+    Thread.sleep(pauseInterval);
+
+    // HTTP call to /api/containers
+    containerResponse = makeHttpCall(containerKeyServiceURL);
+    map = new Gson().fromJson(containerResponse, HashMap.class);
+    linkedTreeMap = (LinkedTreeMap) map.get("data");
+    reconMetadataKeyCount = (long)(double)linkedTreeMap.get("totalCount");
+
+    //verify count of keys
+    Assert.assertEquals(omMetadataKeyCount, reconMetadataKeyCount);
+
+    //verify if Recon Metadata captures vol7/bucket7/key7 info in container7
+    containers = (ArrayList) linkedTreeMap.get("containers");
+    nestedContainerMap = (LinkedTreeMap)containers.get(7);
+    Assert.assertEquals(1.0, nestedContainerMap.get("NumberOfKeys"));
+    Assert.assertEquals(7, (long)(double)
+        nestedContainerMap.get("ContainerID"));
+
+    // HTTP call to /api/task/status
+    omLatestSeqNumber = ((RDBStore) metadataManager.getStore())
+        .getDb().getLatestSequenceNumber();
+
+    taskStatusResponse = makeHttpCall(taskStatusURL);
+    reconLatestSeqNumber = getReconMetadataKeyCount(taskStatusResponse, 2);
+
+    //verify sequence number after Delta Updates
+    Assert.assertEquals(omLatestSeqNumber, reconLatestSeqNumber);
```

Comment:
We do the same set of checks every time -> Key count, specific key presence, sequence number. Can we move the steps into a helper method and call them from the test?